packages/blade/src/components/Input/TextInput/__tests__/TextInput.web.test.tsx

```diff
@@ -0,0 +1,346 @@
+/* eslint-disable react-native-a11y/has-valid-accessibility-descriptors */
+import userEvent from '@testing-library/user-event';
+
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+import { TextInput } from '../';
+import { InfoIcon } from '~components/Icons';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+import assertAccessible from '~src/_helpers/testing/assertAccessible.web';
+
+/**  @TODO: add test case for isLoading once spinner https://github.com/razorpay/blade/pull/685 is merged
+ * 1. either of isLoading or clearbutton should be shown at a time
+ * 2. if isLoading then the input should be disabled
+ */
+describe('<TextInput />', () => {
+  it('should render', () => {
+    const { container } = renderWithTheme(<TextInput label="Enter name" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should display success validation state', () => {
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <TextInput
+        label="Enter name"
+        validationState="success"
+        successText="Success"
+        errorText="Error"
+        helpText="Help"
+      />,
+    );
+
+    const input = getByLabelText(label);
+    const successText = getByText('Success');
+
+    expect(successText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Success');
+    expect(input).toBeValid();
+  });
+
+  it('should display error validation state', () => {
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <TextInput
+        label="Enter name"
+        validationState="error"
+        successText="Success"
+        errorText="Error"
+        helpText="Help"
+      />,
+    );
+
+    const input = getByLabelText(label);
+    const errorText = getByText('Error');
+
+    expect(errorText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Error');
+    expect(input).toBeInvalid();
+  });
+
+  it('should display help text', () => {
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <TextInput label="Enter name" successText="Success" errorText="Error" helpText="Help" />,
+    );
+
+    const input = getByLabelText(label);
+    const helpText = getByText('Help');
+
+    expect(helpText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Help');
+    expect(input).toBeValid();
+  });
+
+  it('should render with icon, prefix, suffix', () => {
+    const { container, getByText } = renderWithTheme(
+      <TextInput
+        label="Enter company website"
+        type="url"
+        placeholder="something"
+        icon={InfoIcon}
+        prefix="https://"
+        suffix=".com"
+      />,
+    );
+    const prefix = getByText('https://');
+    expect(prefix).toBeVisible();
+    const suffix = getByText('.com');
+    expect(suffix).toBeVisible();
+    expect(container).toMatchSnapshot();
```

Comment:
Do we need snapshot for this?

Reply:
it has prefix, suffix, icon so if the icon size changes like we bump up the scale of icons internally then this snapshot will help us that the styling changed